### DISCUSSION TITLE
combine Menu entries [Edit in visual editor] + [Edit in YAML] in Automation/Script editor

### DISCUSSION THREAD

**Discussion by renarena** [home-assistant/frontend#19172] [Automations / scripts / scenes] [answered]
We got new icons for switching between UI and YAML in the card editor some time ago.
But nothing has be done with the menu entries in the automations/script editor.

Only one of the two is useful and you always have to choose the one without any icon (no checkmark). 
They should be combined into one entry only - yaml if editing in UI mode and visual if editing in yaml ;-).

![menu](https://github.com/home-assistant/frontend/assets/4104167/e5e66cd0-6eb6-43e3-bc05-0cc9a8579cfd)

**Comment by renarena** [comment]
We have Disable **OR** Enable - correct from UI View.
Why do we have Edit in visual Editor **AND** Edit in Yaml???
Please show only one Entry to switch to the **OTHER** Editor.

**Comment by marcinbauer-ohf** [comment]
Thank you for pointing this out. I've prepared a solution that can be implemented in Automations, Scripts and Scenes (for scenes please the ongoing work: https://github.com/home-assistant/frontend/pull/22847)

https://www.figma.com/design/XJ50CLDiUkCExBWLckRDtd/HA-16?node-id=2-110&t=O2G9UxWRSGKyBuGo-1

**Comment by silamon** [comment, marked as answer]
Implemented the design through https://github.com/home-assistant/frontend/pull/23142 and additionally for the dashboard area https://github.com/home-assistant/frontend/pull/23143
